apio-consumer/src/main/kotlin/com/liferay/apio/consumer/ApioConsumer.kt

```diff
@@ -15,38 +15,33 @@
 package com.liferay.apio.consumer
 
 import com.github.kittinunf.result.Result
-import com.liferay.apio.consumer.authenticator.ApioAuthenticator
+import com.liferay.apio.consumer.configuration.*
+import com.liferay.apio.consumer.exception.InvalidRequestUrlException
 import com.liferay.apio.consumer.model.Property
 import com.liferay.apio.consumer.model.Thing
-import com.liferay.apio.consumer.request.RequestAuthorization
-import com.liferay.apio.consumer.request.RequestExecutor
-import kotlinx.coroutines.experimental.CommonPool
-import kotlinx.coroutines.experimental.android.UI
-import kotlinx.coroutines.experimental.launch
-import kotlinx.coroutines.experimental.withContext
+import com.liferay.apio.consumer.request.*
+import kotlinx.coroutines.*
 import okhttp3.HttpUrl
 
 /**
  * @author Javier Gamarra
  * @author Paulo Cruz
  */
-class ApioConsumer @JvmOverloads constructor(authenticator: ApioAuthenticator? = null) {
+class ApioConsumer constructor(val defaultHeaders: List<RequestHeader>) {
```

Comment:
I like this change but I'm a bit worried about hiding the authenticator, because it will usually be needed and it's a special common case (and the other headers are rare).

Reply:
Yeah, I agree, maybe we can add an Authorization parameter before and also provide getters and setters as the other headers.